.github/workflows/ci.yml

```diff
@@ -11,209 +11,239 @@ concurrency:
-          NO_TEST_REQUIREMENTS: '${{ matrix.no_test_requirements }}'
-      - if: always()
-        uses: codecov/codecov-action@v3
-        with:
-          directory: empty
-          token: 87cefb17-c44b-4f2f-8b30-1fff5769ce46
-          name: Ubuntu (${{ matrix.python }}${{ matrix.extra_name }})
-          flags: Ubuntu,${{ matrix.python }}
-
-  macOS:
-    name: 'macOS (${{ matrix.python }})'
-    timeout-minutes: 15
-    runs-on: 'macos-latest'
+  Cython:
+    name: "Cython"
+    runs-on: ubuntu-latest
     strategy:
       fail-fast: false
       matrix:
-        python: ['3.8', '3.9', '3.10', 'pypy-3.9-nightly', 'pypy-3.10-nightly']
-    continue-on-error: >-
-      ${{
-        (
-          endsWith(matrix.python, '-dev')
-          || endsWith(matrix.python, '-nightly')
-        )
-        && true
-        || false
-      }}
+        cython: ['==0.29.23', '==0.29.26', '==0.29.30', '<3', '>=3']
+        python: ['3.8', '3.10', '3.11', '3']
     steps:
       - name: Checkout
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
       - name: Setup python
-        uses: actions/setup-python@v4
-        with:
-          python-version: ${{ fromJSON(format('["{0}", "{1}"]', format('{0}.0-alpha - {0}.X', matrix.python), matrix.python))[startsWith(matrix.python, 'pypy')] }}
-          cache: pip
-          cache-dependency-path: test-requirements.txt
-      - name: Run tests
-        run: ./ci.sh
-      - if: always()
-        uses: codecov/codecov-action@v3
-        with:
-          directory: empty
-          token: 87cefb17-c44b-4f2f-8b30-1fff5769ce46
-          name: macOS (${{ matrix.python }})
-          flags: macOS,${{ matrix.python }}
-
-  # run CI on a musl linux
-  Alpine:
-    name: "Alpine"
-    runs-on: ubuntu-latest
-    container: alpine
-    steps:
-      - name: Checkout
-        uses: actions/checkout@v3
-      - name: Install necessary packages
-        # can't use setup-python because that python doesn't seem to work;
-        # `python3-dev` (rather than `python:alpine`) for some ctypes reason,
-        # `nodejs` for pyright (`node-env` pulls in nodejs but that takes a while and can time out the test).
-        run: apk update && apk add python3-dev bash nodejs
-      - name: Enter virtual environment
-        run: python -m venv .venv
-      - name: Run tests
-        run: source .venv/bin/activate && ./ci.sh
-      - if: always()
-        uses: codecov/codecov-action@v3
+        uses: actions/setup-python@v5
         with:
-          directory: empty
-          token: 87cefb17-c44b-4f2f-8b30-1fff5769ce46
-          name: Alpine
-          flags: Alpine,3.12
+            python-version: '${{ matrix.python }}'
+            cache: pip
+      - name: install cython
+        run: |
+          # setuptools is needed to get distutils on 3.12
+          python -m pip install --upgrade pip setuptools
+          python -m pip install "cython${{ matrix.cython}}"
+      - name: install trio
+          # use -e?
```

Comment:
I think editable installs isn't the normal case and those are weird:tm:

So I think it's better to test non-editable installs of trio

Reply:
I was randomly using Black's CI as a reference: https://github.com/search?q=repo%3Apsf%2Fblack+%22python+-m+pip+install+-e%22&type=code
I have no clue why they're doing `-e` tho

Reply:
meh we'll find out if we need it